cpp/CMakeLists.txt

```diff
@@ -637,19 +604,59 @@ if(compiled IN_LIST raft_FIND_COMPONENTS)
 endif()
 ]=]
 )
+set(raft_components compiled distributed)
```

Comment:
I would recommend that we get #1154 merged first since it should simplify the component management in this PR.

Reply:
@vyasr that's indeed the plan. I've merged that branch into this PR in the meantime because these changes won't work as intended otherwise.